packages/blocks/src/api/test/registration.js

```diff
@@ -337,6 +337,42 @@ describe( 'blocks', () => {
 				expect( console ).toHaveErroredWith( 'Block settings must be a valid object.' );
 				expect( block ).toBeUndefined();
 			} );
+
+			it( 'should apply the blocks.registerBlockType filter to each of the deprecated settings as well as the main block settings', () => {
+				const blockSettingsWithDeprecations = {
+					...defaultBlockSettings,
+					deprecated: [
+						{
+							save() {
+								return 1;
+							},
+						},
+						{
+							save() {
+								return 2;
+							},
+						},
+					],
+				};
+
+				addFilter( 'blocks.registerBlockType', 'core/blocks/without-title', ( settings ) => {
```

Comment:
Should we remove the filter after the test to ensure it doesn't mess with other tests?

Reply:
It's already handled in an `afterEach` on line 317.